transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala

```diff
@@ -151,38 +180,48 @@ object HcaPipelineBuilder extends PipelineBuilder[Args] {
     entityType: String,
     fileName: String,
     metadata: Msg,
-    descriptor: Msg
-  ): Msg = {
-    val (entityId, entityVersion) = getEntityIdAndVersion(fileName)
-    // put values in the form we want
-    Obj(
-      Str(s"${entityType}_id") -> Str(entityId),
-      Str("version") -> Str(entityVersion),
-      Str("content") -> Str(encode(metadata)),
-      Str("crc32c") -> descriptor.read[Msg]("crc32c"),
-      Str("descriptor") -> Str(encode(descriptor))
-    )
-  }
+    descriptor: Msg,
+    inputPrefix: String
+  ): Option[Msg] =
+    getEntityIdAndVersion(fileName, inputPrefix, entityType).map {
+      case (entityId, entityVersion) =>
+        Obj(
+          Str(s"${entityType}_id") -> Str(entityId),
+          Str("version") -> Str(entityVersion),
+          Str("content") -> Str(encode(metadata)),
+          Str("crc32c") -> descriptor.tryRead[Msg]("crc32c").getOrElse {
+            MissingPropertyError(
```

Comment:
Yeah this was bothering me too; ooo the filter idea is good! I def want it to filter this out upstream because yeah we are hitting the same thing twice. I might as well just do it now